src/TLabel.cpp

```diff
@@ -104,7 +124,20 @@ void TLabel::mousePressEvent(QMouseEvent* event)
 
 void TLabel::mouseDoubleClickEvent(QMouseEvent* event)
```

Comment:
💭 Note that (like in the Editor class where I suspect it provokes some side-effects we do not properly understand) a double-click is bracketed by __two__ press events (we could simplify the API to only pass the first mouserPressEvent IMVHO and suppress the second one, the one _after_ a double-click)...

Reply:
I saw this from the Qt documentation. Figured at the very least it was out of the scope of the PR, but wasn't sure anyways what to do with it. The only solution I have is very *extensive*... make a TWidget class that is derived from QWidget, which handles those events. Then have all our QWidget derived classes derive TWidget instead.

You guys are much more experienced with this than me though, so I'll leave the brainstorming to y'all :D.

Reply:
What I would probably do is set a flag in the double click event that was detected in the single click (press) event to exit from it without doing anything more (after resetting the flag).  This would filter out the second instance of the click/press event so instead of the underlying code seeing "click" - "double-click" - "click" in sequence it would only see "click" - "double-click"...

... mind you it might be worth checking to see if there was also a double "release" event along with the "press" ones 🙁  - and it may be too late anyhow - these are not new lua callback/events are they, in which case suppressing the second click would be a change in the API and thus __verboten__!

Reply:
Ahh right... there could potentially be scripts out there that check double clicks by timing subsequent "single" clicks. I hadn't considered that!

Reply:
I don't think we need to do this, not unless it becomes necessary.